src/axom/klee/Geometry.hpp

```diff
@@ -69,21 +78,169 @@ class Geometry
+   *
+   * \internal TODO: Is this the simplex requirement overly restrictive?
+   */
+  Geometry(const TransformableGeometryProperties &startProperties,
+           const axom::primal::Sphere<double, 3> &sphere,
+           axom::IndexType levelOfRefinement,
+           std::shared_ptr<GeometryOperator const> operator_);
+
+  /**
+   * Create a volume-of-revolution (VOR) Geometry object.
+   *
+   * \param startProperties the transformable properties before any
+   * operators are applied
+   * \param discreteFunction Discrete function describing the surface
+   *        of revolution.
+   * \param vorBase Coordinates of the base of the VOR.
+   * \param vorDirection VOR axis, in the direction of increasing z.
+   * \param levelOfRefinement Number of refinement levels to use for
+   *        discretizing the sphere.
+   * \param operator_ a possibly null operator to apply to the geometry.
+   *
+   * The \c discreteFunction should be an Nx2 array, interpreted as
+   * (z,r) pairs, where z is the axial distance and r is the radius.
+   * The \c vorBase coordinates corresponds to z=0.
+   * \c vorAxis should point in the direction of increasing z.
+   *
+   * \internal TODO: Is this the simplex requirement overly restrictive?
+   */
+  Geometry(const TransformableGeometryProperties &startProperties,
+           const axom::Array<double, 2> &discreteFunction,
+           const Point3D &vorBase,
+           const Vector3D &vorDirection,
+           axom::IndexType levelOfRefinement,
+           std::shared_ptr<GeometryOperator const> operator_);
+
+  /**
+   * Create a planar Geometry object.
+   *
+   * \param startProperties the transformable properties before any
+   * operators are applied
+   * \param tet Tetrahedron
+   * \param operator_ a possibly null operator to apply to the geometry.
+   *
+   * The space on the positive normal side of the plane is considered
+   * "inside the shape".
+   */
+  Geometry(const TransformableGeometryProperties &startProperties,
+           const axom::primal::Plane<double, 3> &plane,
+           std::shared_ptr<GeometryOperator const> operator_);
+
+  /**
+   * @brief Get the format in which the geometry was specified.
+   *
+   * The format is determined by the constructor used.
+   * Values are:
+   * - "c2c" = C2C file
+   * - "proe" = ProE file
+   * - "blueprint-tets" = Blueprint tetrahedral mesh in memory
+   * - "tet3D" = 3D tetrahedron (4 points)
+   * - "sphere3D" = 3D sphere, as \c primal::Sphere<double,3>
+   * - "vor3D" = 3D volume of revolution.
+   * - "cone3D" = 3D cone, as \c primal::Cone<double,3>
+   * - "cylinder3D" = 3D cylinder, as \c primal::Cylinder<double,3>
+   * - "hex3D" = 3D hexahedron (8 points)
+   * - "plane3D" = 3D plane
    *
    * \return the format of the shape
+   *
+   * TODO: Depending on the specified geometry, some members are not
+   * used.  It may clarify make each supported geometry a subclass.
```

Comment:
```suggestion
   * used.  It may clarify if we make each supported geometry a subclass.
```

Are comments like this captured in issues so they don't get lost?

Reply:
Issue https://github.com/LLNL/axom/issues/1445